src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -750,6 +750,13 @@ public void update() {
         actions.put(Actions.REMOVE_FROM_GROUP, new GroupAddRemoveDialog(this, false, false));
         actions.put(Actions.MOVE_TO_GROUP, new GroupAddRemoveDialog(this, true, true));
 
+        actions.put(Actions.INTERSECTION_PREF_GROUP, (BaseAction) () -> {
+            System.out.println("blubb");
+            boolean state = Globals.prefs.getBoolean(JabRefPreferences.GROUP_INTERSECT_SELECTIONS);
+            state = !state;
```

Comment:
This code is hard to understand and not clear on the first sight.
Better create two variables: e.g. union and intersection and then store the corresponding value

Reply:
Ideally you would have an enum with two states, e.g. GroupState... but in this case this might be an overkill.